opentech/apply/projects/views.py

```diff
@@ -106,25 +106,54 @@ def form_valid(self, form):
         return response
 
 
+@method_decorator(staff_required, name='dispatch')
+class UploadDocumentView(DelegatedViewMixin, CreateView):
+    context_name = 'document_form'
+    form_class = UploadDocumentForm
+    model = Project
+
+    def form_valid(self, form):
+        project = self.kwargs['object']
+        form.instance.project = project
+        response = super().form_valid(form)
+
+        messenger(
+            MESSAGES.UPLOAD_DOCUMENT,
+            request=self.request,
+            user=self.request.user,
+            source=project,
+            title=form.instance.title
+        )
+
+        return response
+
+
 class AdminProjectDetailView(ActivityContextMixin, DelegateableView, DetailView):
     form_views = [
         CommentFormView,
         CreateApprovalView,
         RejectionView,
         SendForApprovalView,
         UpdateLeadView,
+        UploadDocumentView,
     ]
     model = Project
     template_name_suffix = '_admin_detail'
 
     def get_context_data(self, **kwargs):
+        categories = (DocumentCategory.objects.exclude(packet_files__project=self.object)
```

Comment:
This wont cut it. It ignores the minimum required aspect of the data.

We should choose one of the following:
1. iterate over the relations in a dict to calculate what is left to add 
2. do nothing and list all required

1. is probably the most value in this instance i think?

There is probably also a query we could do to calculate it, but that sounds ott.

Reply:
Fixed in 6497d047